File: trivially/src/main/java/trivially/ep/i2/Mult.java
package trivially.ep.i2;

public abstract interface Mult<V> extends trivially.ep.m7.Mult<V>, Exp<V> {

    public abstract Exp<V> getLeft();

    public abstract Exp<V> getRight();

    public default Exp<V> simplify() {
        return this.convert(trivially.ep.m7.Mult.super.simplify());
    }

    public default Exp<V> multby(trivially.ep.Exp<V> other) {
        return this.convert(trivially.ep.m7.Mult.super.multby(other));
    }

    public default Exp<V> powby(trivially.ep.Exp<V> other) {
        return this.convert(trivially.ep.m7.Mult.super.powby(other));
    }
}

File: coco_csharp/CoCo/EP/M7/Mult.cs
namespace EP.M7 {
    public interface Mult<FT> : Exp<FT>, EP.M6.Mult<FT> {
        EP.Exp<FT> Exp<FT>.powby(EP.Exp<FT> other) {
            return mult(convert(getLeft()).powby(convert(other)), convert(getRight()).powby(convert(other)));
        }
    }
}